### PR TITLE
fix: Resolved warnings false detection in ColorScheme definition

### LINKED ISSUE #65: Exclude `avoid_hardcoded_japanese` for entire test directory, not just `*_test.dart`
I want to exclude `avoid_hardcoded_japanese` from test code.
Currently exclude only `*_test.dart`, but I want to exclude all test files such as test_utils.dart.

### LINKED ISSUE #67: `avoid_hardcoded_color` warnings false detection in ColorScheme definition
**Describe the bug**
I get a warning `avoid_hardcoded_color` when defining ColorScheme

**To Reproduce**
```dart
ColorScheme get _darkColorScheme => const ColorScheme(
      brightness: Brightness.dark,
      primary: Color.fromRGBO(xxx, xxx, xxx, x), // `avoid_hardcoded_color` warnings here
);
```

**Expected behavior**
ColorScheme definition has not warnings

### PR DESCRIPTION
**Pull request by boywithdv**
## 🔗 Related Issues

- closes #67 

## 🙌 What's Done

- Resolving the bug that warnings are also displayed in colorScheme definitions.

## ✍️ What's Not Done

- 

## 🖼️ Image Differences

| Before    | After     |
| --------- | --------- |
| __image__ | __image__ |

## 🤼 Desired Review Method

- [ ] Correction Commit
- [ ] Pair programming

> [!NOTE]
> It is possible that a reviewer's will may cause a method to be implemented that is not selected.

## Pre-launch Checklist
- [ ] I have reviewed my own code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I updated/added relevant documentation (doc comments with ///).

### CODE REVIEW

File: packages/altive_lints/lib/src/lints/avoid_hardcoded_color.dart [hunk 1]
```diff
@@ -48,7 +49,7 @@ class AvoidHardcodedColor extends DartLintRule {
     context.registry.addInstanceCreationExpression((node) {
       final typeName = node.staticType?.getDisplayString();
 
-      if (typeName == 'Color') {
+      if (typeName == 'Color' && !_isInsideColorScheme(node)) {
         reporter.atNode(node, _code);
       }
```
Comment (k-nkmr) [inline packages/altive_lints/lib/src/lints/avoid_hardcoded_color.dart:54]: badge 

I suggest moving the `_isInsideColorScheme(node)` check to the beginning of this method. By using an early return here, we can avoid redundant checks and improve the code’s readability. 

If `_isInsideColorScheme(node)` returns `true`, we immediately exit, which makes it clear that further processing only applies to nodes outside of a `ColorScheme`. This also eliminates the need to call `_isInsideColorScheme()` multiple times, resulting in a slight performance improvement.

File: packages/altive_lints/lib/src/lints/avoid_hardcoded_color.dart [hunk 2]
```diff
@@ -48,7 +49,7 @@ class AvoidHardcodedColor extends DartLintRule {
     context.registry.addInstanceCreationExpression((node) {
       final typeName = node.staticType?.getDisplayString();
```
Comment (k-nkmr) [inline packages/altive_lints/lib/src/lints/avoid_hardcoded_color.dart]:
```suggestion
    context.registry.addInstanceCreationExpression((node) {
     if (_isInsideColorScheme(node)) {
          return;
      }
      final typeName = node.staticType?.getDisplayString();
      
```

badge 
I suggest moving the _isInsideColorScheme(node) check to the beginning of this method. By using an early return here, we can avoid redundant checks and improve the code’s readability.

If _isInsideColorScheme(node) returns true, we immediately exit, which makes it clear that further processing only applies to nodes outside of a ColorScheme. This also eliminates the need to call _isInsideColorScheme() multiple times, resulting in a slight performance improvement.

Comment (boywithdv) [reply packages/altive_lints/lib/src/lints/avoid_hardcoded_color.dart]: Thank you very much.
Indeed, it looks like isInsideColorScheme should be checked first.

**Review by k-nkmr — APPROVED** [review]
LGTM

**Pull request merged**